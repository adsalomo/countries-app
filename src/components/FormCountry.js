import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { useForm } from '../hooks/useForm';

export const FormCountry = ({ setCountries }) => {
    const [propertiesForm, handleOnChange, handleReset] = useForm({
        country: '',
        peoples: '',
    });
    const [errorMessage, setErrorMessage] = useState(null);
    const { country, peoples } = propertiesForm;

    const handleSave = (e) => {
        e.preventDefault();

        //console.log(country, peoples);

        if (!country || country.length <= 0 || !peoples || peoples.length <= 0) {
            setErrorMessage('PaÃ­s y Habitantes es requerido');
            return;
        }

        console.log('Hola mundo', country, peoples);
        setCountries((countries) => [{ country, peoples: Number(peoples) }, ...countries]);
        handleReset();
        setErrorMessage(null);
    }

    return (
        <>
            <form onSubmit={handleSave}>
                <div className="form-group">
                    <label>Pais</label>
                    <input onChange={handleOnChange} value={country ? country : ''} name="country" className="form-control" type="text" placeholder="Escriba pais" autoComplete="off" />
                </div>
                <div className="form-group">
                    <label>Habitantes</label>
                    <input onChange={handleOnChange} value={peoples ? peoples : ''} name="peoples" className="form-control" type="number" placeholder="Escriba numero" />
                </div>
                <button className="btn btn-primary btn-block" type="submit">Nuevo pais</button>
                {errorMessage && <span className="text-danger">{`*${errorMessage}`}</span>}
            </form>
        </>
    )
}

FormCountry.propTypes = {
    setCountries: PropTypes.func.isRequired,
}


