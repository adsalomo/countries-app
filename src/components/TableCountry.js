import React from 'react';
import PropTypes from 'prop-types';

export const TableCountry = ({ countries }) => {
    console.log('TableCountry');

    return (
        <table className="table mt-3">
            <thead>
                <tr>
                    <th>#</th>
                    <th>Nombre</th>
                    <th>NÃºmero habitantes</th>
                </tr>
            </thead>
            <tbody>
                {
                    countries && countries.length > 0
                    && countries.map(({ country, peoples }, i) => {
                        return <tr key={i}>
                            <td>{i + 1}</td>
                            <td>{country}</td>
                            <td>{peoples}</td>
                        </tr>
                    })
                }
            </tbody>
        </table>
    )
}

TableCountry.propTypes = {
    countries: PropTypes.array.isRequired,
}
