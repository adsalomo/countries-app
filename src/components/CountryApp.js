import React, { useEffect, useState } from 'react';
import { getCountries } from '../helpers/getCountries';
import { FormCountry } from './FormCountry';
import { TableCountry } from './TableCountry';

export const CountryApp = () => {
    const [countries, setCountries] = useState([]);

    useEffect(() => {
        getCountries().then(resp => {
            setCountries(resp);
        });
    }, []);


    return (
        <>
            <h1>Countries App</h1>
            <hr />

            <FormCountry
                setCountries={setCountries}
            />

            <TableCountry countries={countries} />
        </>
    )
}
