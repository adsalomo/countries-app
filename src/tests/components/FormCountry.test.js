import { shallow } from 'enzyme'
import { FormCountry } from '../../components/FormCountry';

import '@testing-library/jest-dom';

describe('Prueba componente FormCountry', () => {

    const setCountries = jest.fn();
    let wrapper = shallow(<FormCountry setCountries={setCountries} />);

    beforeEach(() => {
        wrapper = shallow(<FormCountry setCountries={setCountries} />);
    });

    test('Debe traer la foto del componente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Simula agregar nuevo pais', () => {
        const inputs = wrapper.find('input');
        //console.log(inputs.at(0).props());
        //console.log(inputs.at(1).props());

        const inputCountry = inputs.at(0);

        inputCountry.simulate('change', {
            target: {
                name: 'country',
                value: 'Canada',
            }
        });

        const inputPeoples = inputs.at(1);

        inputPeoples.simulate('change', {
            target: {
                name: 'peoples',
                value: 1000,
            }
        });

        // Formulario
        wrapper.find('form').simulate('submit', {
            preventDefault() { }
        });

        //expect(setCountries).toHaveBeenCalled();
        //expect(setCountries).toHaveBeenCalledWith(expect.any(Function));

    });


});
