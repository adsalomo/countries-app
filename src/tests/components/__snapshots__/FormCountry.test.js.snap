// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba componente FormCountry Debe traer la foto del componente 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label>
        Pais
      </label>
      <input
        autoComplete="off"
        className="form-control"
        name="country"
        onChange={[Function]}
        placeholder="Escriba pais"
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Habitantes
      </label>
      <input
        className="form-control"
        name="peoples"
        onChange={[Function]}
        placeholder="Escriba numero"
        type="number"
        value=""
      />
    </div>
    <button
      className="btn btn-primary btn-block"
      type="submit"
    >
      Nuevo pais
    </button>
  </form>
</Fragment>
`;
